resources/istio/files/istio-proxy-reset.sh

```diff
@@ -1,40 +1,151 @@
 #!/usr/bin/env bash
 
-#We expect all sidecars to be in this version
-expectedIstioProxyVersion="${EXPECTED_ISTIO_PROXY_IMAGE:-eu.gcr.io/kyma-project/external/istio/proxyv2:1.7.4-distroless}"
-#We use this image prefix to detect if there's an istio proxy in the Pod. If you have different prefixes (shouldn't be the case), just define several jobs.
-istioProxyImageNamePrefix="${COMMON_ISTIO_PROXY_IMAGE_PREFIX:-eu.gcr.io/kyma-project/external/istio/proxyv2}"
+trap cleanup EXIT SIGTERM
 
-dryRun="${DRY_RUN:-false}"
+cleanup() {
+    if [[ -n "${REMOVE_PODS_FILE}" ]]; then
+        echo
+        echo "Removing temporary file with pods data: ${REMOVE_PODS_FILE}"
+        rm "${REMOVE_PODS_FILE}"
+    fi
+}
+
+# Retries a command on failure.
+# $1 - the max number of attempts
+# $2... - the command to run
+retry() {
+    local -r -i max_attempts="$1"; shift
+    local -r cmd="$@"
+    local -i attempt_num=1
+
+    until $cmd
+    do
+        if (( attempt_num == max_attempts ))
+        then
+            echo "Attempt $attempt_num failed and there are no more attempts left!"
+            exit $exitCode
+        else
+            echo "Attempt $attempt_num failed! Trying again in $attempt_num seconds..."
+            sleep $(( attempt_num++ ))
+        fi
+    done
+}
+
+# Deletes a pod
+deletePod() {
+    local namespace=$1
+    local podName=$2
 
-for NS in $(kubectl get ns -l kyma-project.io/created-by=e2e-upgrade-test-runner -o name | cut -d '/' -f2); do
     if [[ "${dryRun}" == "false" ]]; then
-        kubectl delete rs -n "${NS}" --all
+        echo "    Deleting pod: ${namespace}/${podName}"
+        retry "${RETRIES_COUNT}" kubectl -n "${namespace}" delete pod "${podName}"
+        sleep 1
```

Comment:
why do we need that sleep?

Reply:
It is here to slow down Pod deletion. Otherwise, if the pods are deleted "too fast", all the Pods of some ReplicationController might be deleted before the controller spawns new ones and user face downtime.
Essentially, it's a "poor man's rollout" attempt. Remember that we delete Pods for objects where no "rollout" command exists. The problem is, proper value for this depends on the environment. In some it may be not necessary at all, in some it may be 5 seconds? Maybe we should make it configurable with a default value of zero?